src/blockdata/block.rs

```diff
@@ -95,24 +94,25 @@ impl Block {
     }
 
     /// compute witness commitment for the transaction list
-    pub fn compute_witness_commitment (witness_root: &sha256d::Hash, witness_reserved_value: &[u8]) -> sha256d::Hash {
-        let mut encoder = sha256d::Hash::engine();
+    pub fn compute_witness_commitment (witness_root: &WitnessMerkleRoot, witness_reserved_value: &[u8]) -> WitnessCommitment {
+        let mut encoder = WitnessCommitment::engine();
         witness_root.consensus_encode(&mut encoder).unwrap();
         encoder.input(witness_reserved_value);
-        sha256d::Hash::from_engine(encoder)
+        WitnessCommitment::from_engine(encoder)
     }
 
     /// Merkle root of transactions hashed for witness
-    pub fn witness_root(&self) -> sha256d::Hash {
-        let mut txhashes = vec!(sha256d::Hash::default());
+    pub fn witness_root(&self) -> WitnessMerkleRoot {
+        let mut txhashes = vec!(Txid::default());
         txhashes.extend(self.txdata.iter().skip(1).map(|t|t.bitcoin_hash()));
```

Comment:
It's not clear to my why this function that must compute witness commitment uses `bitcoin_hash` returning `Txid` whence we need `Wtxid`s